postgres/manage.sls

```diff
@@ -18,7 +20,11 @@ include:
 # Ensure that Salt is able to use postgres modules
 
 postgres-reload-modules:
+{%- if needs_client_binaries or postgres.manage_force_reload_modules %}
   test.succeed_with_changes:
+{%- else %}
+  test.succeed_without_changes:
+{%- endif %}
```

Comment:
@RobRuana This seems like an obvious fix, but that wouldn't work.
Each state generated below explicitly subscribed on changes in `postgres-reload-modules`. So if you later change PG entities in the Pillar, it will not have any effect.
The only way I see to overcome this: you need to put conditional in `format_state` macro to depend on modules reload only when `postgres.manage_force_reload_modules` set `True`.

Reply:
@vutny, yes, you are correct that this breaks any state using the `format_state()` macro.

At first I couldn't understand why `onchanges: postgres-reload-modules` was being used instead of `require: postgres-reload-modules`. But then I realized that many instances of `format_state()` are already using the `require` conditional, so there would be conflicting `require` keys when the state is rendered.

Correct me if I'm wrong, but I believe `onchanges` is being used to force correct ordering of the states, not because we want every `format_state()` to run when there are changes in `postgres-reload-modules`.

If that's correct, then we should be able to achieve the same result by using `watch` instead of `onchanges` in `format_state()`. I've made that change. Please let me know if you think that works.